utils/src/main/scala/za/co/absa/enceladus/utils/udf/UDFLibrary.scala

```diff
@@ -34,65 +34,74 @@ object UDFLibrary {
 
   private var isRegistered = false
 
-  private def registerUdf()(implicit spark: SparkSession) {
+  private var registry: Set[Int] = Set.empty
```

Comment:
Wouldn't it be easier to use a existing synchronized set/map. E.g. `java.util.concurrent.ConcurrentHashMap[K, Unit].putIfAbsent`?

Reply:
Because I didn't know the class. 👼